### PR TITLE
Fix 'undo' after ':TerraformFmt'

### PR DESCRIPTION
**Pull request by dimbleby**

### CODE REVIEW

File: autoload/terraform.vim [hunk 1]
```diff
@@ -3,6 +3,9 @@ function! terraform#fmt()
     return
   endif
   let l:curw = winsaveview()
+  " Make a fake change so that the undo point is right.
+  normal! ix
+  normal! x
```
Comment (svanharmelen) [inline autoload/terraform.vim]: This looks pretty nasty. I really prefer not too do this... Why would we want to introduce this kind of nastiness if we already know a solution which is used in other places as well?

Comment (svanharmelen) [reply autoload/terraform.vim]: I rebased #101 and prefer to use that solution as the same solution is already used (so tested) in other well known plugins like `vim-go` and `vim-hclfmt`.

Comment (dimbleby) [reply autoload/terraform.vim]: Yeah, I'm in two minds.  

I agree that this `ix`, `x` trickery is not nice.  

On the other hand the other implementation is also unsatisfactory to my mind - it's kinda complicated, it unexpectedly writes the file to disk, and it has weirdness of its own (what is `let &syntax = &syntax`?!)

**Comment by svanharmelen** [comment]
@dimbleby unfortunately this doesn't solve the problem 😞 The problem is a bit less annoying as now the cursor doesn't move to the top of the file anymore. But pressing `undo` still doesn't work as expected when you try to undo something after first having received a fmt error.

Now the first time you press `u` nothing happens at all. Only the second time it actually undo's the latest change. This is still annoying and undesirable IMO. So can we please use the other solution instead?

It's not that I'm married to that other solution (I didn't even write it), but it "just works" without any side effects.

Thanks!

**Comment by dimbleby** [comment]
@svanharmelen I'm not following the problem now - so far as I can see everything is working fine for me, whether formatting give an error or not.  

Can you spell out for me a sequence that's not good for you?

**Comment by svanharmelen** [comment]
Sure, it's pretty straight forward...

1. Make a syntax error in your config and execute `:w`
2. Fix the syntax error and execute `:w` again
3. Execute `:undo` and see that nothing changes (no undo is being done) but the buffer is marked as changed
4. Execute `:undo` again to undo the last change

So it now takes 2 `:undo`'s to undo the last change. See this small screen capture for more info:

Kapture 2019-08-06 at 21 06 10

**Comment by dimbleby** [comment]
I reckon what you're seeing is that 
- the formatting changed nothing
- so undoing the formatting changes nothing

Try again with the file syntactically correct but badly formatted, and I expect you'll see that the first undo undoes the formatting, and the second undo undoes the previous change.

I think that having a separate undo point for the formatting is probably correct behaviour - perhaps you don't agree with that?

Certainly I don't want to get into the game of trying to figure out whether the formatting actually changed anything, and changing the undo sequence accordingly...

**Comment by dimbleby** [comment]
Also, I believe that the alternative implementation that we're discussing has the same behaviour - doesn't it?

**Comment by svanharmelen** [comment]
Yes, you are correct... It's indeed a result of the formatting action and its common behavior for when files are formatted on (pre) save. So I stand corrected and think I just didn't pay that much attention to this behavior before.

So all good I guess 👍

**Comment by dimbleby** [comment]
No worries!

**Pull request merged**